lib/portlayer/network/context.go

```diff
@@ -85,18 +89,23 @@ func NewContext(bridgePool net.IPNet, bridgeMask net.IPMask, config *Configurati
 		defaultBridgePool: NewAddressSpaceFromNetwork(&bridgePool),
 		scopes:            make(map[string]*Scope),
 		containers:        make(map[string]*Container),
+		PortGroups:        decodeNetworkReferences(&vctx, sess, config),
 	}
 
-	s, err := ctx.newBridgeScope(uid.New(), DefaultBridgeName, nil, net.IPv4(0, 0, 0, 0), nil, &IPAM{})
+	return ctx, nil
+}
+
+func (c *Context) Init() error {
```

Comment:
So it's possible to set any value of the context, before actually initialize any in memory cache, actually to initialize the scopes.
Otherwise, the NewContext method will resolve portgroups before create scopes, which means unit test will need govmomi connection. 
It's ok to change unit test to connect vsphere, but as the majority function is not related to vsphere, I assume this will make it easier.